tensorflow/core/kernels/xent_op.cc

```diff
@@ -58,6 +86,17 @@ class SoftmaxXentWithLogitsOp : public OpKernel {
                                         "2-dimensional, or broadcasted to be "
                                         "2-dimensional"));
 
+    if (std::is_same<Device, GPUDevice>::value) {
```

Comment:
Is the CPU implementation deterministic?

Reply:
We just confirmed that the CPU implementation is deterministic (thanks @wenscarl). I'm considering add tests to prove/confirm/ensure that in a future PR.